notebooks/2.modeling/nbconverted/2.modeling.py

```diff
@@ -90,208 +162,160 @@
 }
 
 
-# Loading training data splits
-#
+# ### Training both non shuffled and shuffled models with feature selected cell injury profiles (not aligned with JUMP)
 
-# In[4]:
-
-
-# loading injury codes
-injury_codes = load_json_file(feature_dir / "injury_codes.json")
-
-# load share feature space data
-feature_space = load_json_file(feature_space_path)
-shared_features = feature_space["features"]
-
-# loading training data splits
-X_train = pd.read_csv(X_train_path)
-X_test = pd.read_csv(X_test_path)
-y_train = pd.read_csv(y_train_path)
-y_test = pd.read_csv(y_test_path)
-
-
-# splitting meta and feature column names
-_, feat_cols = split_meta_and_features(X_train)
-
-# checking if the feature space are identical (also looks for feature space order)
-assert check_feature_order(
-    ref_feat_order=shared_features, input_feat_order=X_test.columns.tolist()
-), "Feature space are not identical"
+# In[6]:
 
-# display data split sizes
-print("X training size", X_train.shape)
-print("X testing size", X_test.shape)
-print("y training size", y_train.shape)
-print("y testing size", y_test.shape)
 
+# if trained model exists, skip training
+if fs_model_path.exists():
+    fs_best_model = joblib.load(fs_model_path)
 
-# ## Training and Evaluating Multi-class Logistic Model with original dataset split
-#
+# train model and save
+else:
+    fs_best_model = train_multiclass(
+        fs_X_train_df,
+        fs_y_train_df,
+        param_grid=param_grid,
+        seed=seed,
+        cv_results_outpath=fs_model_cv_results_path,
+    )
+    joblib.dump(fs_best_model, fs_model_path)
 
-# In[5]:
 
+# Training shuffled model
+# shuffle feature space
+fs_shuffled_X_train = shuffle_features(fs_X_train_df, features=fs_feats, seed=seed)
 
-# setting model path
-model_path = modeling_dir / "multi_class_model.joblib"
+# checking if the shuffled and original feature space are the same
+assert not fs_X_train_df.equals(fs_shuffled_X_train), "DataFrames are the same!"
```

Comment:
There is a `not` here, so will this confirm that they are different. Also, don't we expect them to be different? Why not check the columns are the same only? (maybe that is what `.equals()` does?)

Reply:
We expect `X_train` to be different from `shuffled_X_train`. The purpose of this check is to ensure that the data was indeed shuffled before training the model. Your question about `.equals()`, it checks both the values and the structure of the DataFrame, including the index, column labels, and the actual data.

Regarding the use of `not` in the assert statement: assertions evaluate to `True`, so if the DataFrames are not equal, `assert not` ensures that the mismatch returns `True` and does not trigger the assertion failure. However, this logic can be confusing at a glance.

Now, looking at this in retrospect, I agree it could be clearer. I plan to rewrite this using exceptions to improve readability.

Reply:
I have updated this from 
```python
assert not fs_shuffled_X_train.equals(fs_X_train_df), "DataFrames are the same!"
```

to 
```python
if fs_shuffled_X_train.equals(fs_X_train_df):
    raise ValueError("Error: DataFrames are the same,")
```